### ISSUE TITLE
composer.json error

### ISSUE THREAD

**Issue by damien-synolia** [hipay/hipay-fullservice-sdk-magento2#87]
Hello HiPay Team,

We just tried to install this component and we saw an error in composer.json.

`"autoload": {
        "psr-4": {
            "HiPay\\": "tests/functional/tests/app/HiPay/"
        }
    },`

You never declare the module. See the following example from Magento Catalog Module.

`"autoload": {
[...]
        "psr-4": {
            "Magento\\Catalog\\": ""
        }
    }`

You have to specify how to load your module. So it should be, in your composer.json:

`
     "autoload": {
        "psr-4": {
            "HiPay\\FullserviceMagento\\": ""
        }
    },`

Without this code, Magento is unable to load your module (php bin/magento module:status do not show Hipay).

Damien.

**Comment by jprotin** [comment]
Hi Damien,

We are looking at this quickly.

Thank you very much !

Johan

**Comment by legithubdeaymeric** [comment]
Hello Damien,

We use the "extra-map" with a registration.php to declare the component and the module should be loaded without problems normally.

Could you tell us which version of magento and the Hipay module you are using?

Thank you

**Comment by damien-synolia** [comment]
Hello,

We are working an Magento Enterprise 2.1.6. 

Damien.

**Comment by legithubdeaymeric** [comment]
We do not reproduce the problem but we are bringing a correction for the next release 1.0.9 that will be available tomorrow.

Did you install the module in the meantime?
Do not hesitate to contact us if you still have problems.

**Comment by damien-synolia** [comment]
Hello !

I updated the module via composer but still have the problem... I have to add it to code to make it work.

Damien.

**Comment by legithubdeaymeric** [comment]
Is the module installed in the "app/code/HiPay/FullserviceMagento/" directory ?
If so, have you realized the installation with the right rights ?

**Comment by damien-synolia** [comment]
Hi,

We tried to install the module via composer (in vendor). But without success. So I finally decide to copy the whole code from vendor to app/code. This way, it works. But it's not the good process. We should let composer do its job.

I have all access rights on my dev machine ;).

Damien.

**Comment by legithubdeaymeric** [comment]
I totally agree with you! Do not do that!

I think you do not use the magento install composer to install your module. Look at :
https://github.com/magento/magento-composer-installer

Do you add well dependency in the magento root composer.json?

**Comment by damien-synolia** [comment]
Hello,

For our modules, we use standard composer json files. I will look to this page for the future, thank you.

For your module, we just have in composer.json : 
`"hipay/hipay-fullservice-sdk-magento2": "^1.0",`

Resulting on this on in the .lock : 

```
{
            "name": "hipay/hipay-fullservice-sdk-magento2",
            "version": "1.1.0",
            "source": {
                "type": "git",
                "url": "https://github.com/hipay/hipay-fullservice-sdk-magento2.git",
                "reference": "cfcbd11965668188bf57e2ceb6c31441a20b3f11"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/hipay/hipay-fullservice-sdk-magento2/zipball/cfcbd11965668188bf57e2ceb6c31441a20b3f11",
                "reference": "cfcbd11965668188bf57e2ceb6c31441a20b3f11",
                "shasum": ""
            },
            "require": {
                "hipay/hipay-fullservice-sdk-php": "^1.0",
                "php": "~5.5.0|~5.6.0|~7.0.0"
            },
            "type": "magento2-module",
            "extra": {
                "map": [
                    [
                        "src/*",
                        "HiPay/FullserviceMagento"
                    ]
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "Apache 2"
            ],
            "description": "The HiPay Fullservice module for Magento 2 is a PHP module which allows you to accept payments in your Magento 2 online store.",
            "time": "2017-05-30T16:17:44+00:00"
        },
        {
            "name": "hipay/hipay-fullservice-sdk-php",
            "version": "1.4.0",
            "source": {
                "type": "git",
                "url": "https://github.com/hipay/hipay-fullservice-sdk-php.git",
                "reference": "a8449353f9b32adeaf263a202a994dfb6761ae26"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/hipay/hipay-fullservice-sdk-php/zipball/a8449353f9b32adeaf263a202a994dfb6761ae26",
                "reference": "a8449353f9b32adeaf263a202a994dfb6761ae26",
                "shasum": ""
            },
            "require": {
                "php": ">=5.3.0"
            },
            "require-dev": {
                "phpdocumentor/phpdocumentor": "^2.8",
                "phpunit/phpunit": "5.7.*"
            },
            "type": "library",
            "autoload": {
                "psr-4": {
                    "HiPay\\Fullservice\\": "lib/HiPay/Fullservice/"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "Apache 2.0"
            ],
            "authors": [
                {
                    "name": "HiPay",
                    "email": "support.tpp@hipay.com",
                    "homepage": "http://www.hipay.com"
                }
            ],
            "description": "The HiPay Enterprise SDK for PHP is a library for developers who want to integrate HiPay Enterprise payment methods to any PHP platform.",
            "homepage": "https://hipay.com/product-enterprise/",
            "keywords": [
                "hipay",
                "payment",
                "payment method"
            ],
            "time": "2017-05-29T13:49:37+00:00"
        },
```

**Comment by legithubdeaymeric** [comment]
Hello Damien, 

Fix in Version 1.1.2
